ovn_k8s/common/kubernetes.py

```diff
@@ -17,22 +17,60 @@
 
 import ovs.vlog
 
+from os import path
 from ovn_k8s.common import exceptions
+from ovn_k8s.common.util import ovs_vsctl
 
+CA_CERTIFICATE = "/etc/openvswitch/k8s-ca.crt"
 vlog = ovs.vlog.Vlog("kubernetes")
 
 
+def _get_api_params():
+    ca_certificate = None
+    api_token = None
+    k8s_api_server = ovs_vsctl("--if-exists", "get", "Open_vSwitch", ".",
+                               "external_ids:k8s-api-server").strip('"')
+    if k8s_api_server.startswith("https://"):
+        if not path.isfile(CA_CERTIFICATE):
+            k8s_ca_crt = ovs_vsctl("--if-exists", "get", "Open_vSwitch",
+                                   ".", "external_ids:k8s-ca-certificate"
+                                   ).strip('"')
+            if k8s_ca_crt:
+                k8s_ca_crt = k8s_ca_crt.replace("\\n", "\n")
+                ca_file = open(CA_CERTIFICATE, 'w+')
+                ca_file.write(k8s_ca_crt)
+                ca_certificate = CA_CERTIFICATE
+        else:
+            ca_certificate = CA_CERTIFICATE
```

Comment:
what bothers me a bit is that this function is called for every single call to the kubernetes.
While ovsdb operations are not expensive, I would still like to avoid doing one for every k8s API call.

To avoid this it will be sufficient to store k8s_api_server in a module-level variable or use a function-level static variable. 

Or - even better imho - do not call ovs-vsctl at all and expect the api server location to be passed to this function (we have it in variables.K8S_API_SERVER).

Is this something you might find useful?